### PR TITLE
Add OneSignal support & dependencies 

### PR DESCRIPTION
**Pull request by aahmad4**
* App users can now receive push notifications via OneSignal

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -5,6 +5,10 @@ plugins {
     id 'dagger.hilt.android.plugin'
 }
 
+plugins {
+    id 'com.onesignal.androidsdk.onesignal-gradle-plugin'
+}
+
```
Comment (aahmad4) [inline app/build.gradle]: @z3bi For some reason, adding this in the `plugins {}` above wasn't working, but this did the trick

File: app/src/main/java/org/raleighmasjid/iar/MainApplication.kt [hunk 2]
```diff
@@ -1,7 +1,21 @@
 package org.raleighmasjid.iar
 
 import android.app.Application
+import com.onesignal.OneSignal
 import dagger.hilt.android.HiltAndroidApp
 
+const val ONESIGNAL_APP_ID = "01fcf852-7b3f-4b53-a733-4cb8241bd193"
+
```
Comment (aahmad4) [inline app/src/main/java/org/raleighmasjid/iar/MainApplication.kt:8]: Should this be some kind of `env` variable? How can we handle this?

File: build.gradle [hunk 3]
```diff
@@ -4,6 +4,7 @@ buildscript {
     }
     dependencies {
         classpath 'com.google.dagger:hilt-android-gradle-plugin:2.40.5'
+        classpath 'gradle.plugin.com.onesignal:onesignal-gradle-plugin:[0.12.10, 0.99.99]'
```
Comment (z3bi) [inline build.gradle]: I think we can just specify an exact version and then update if needed

File: app/build.gradle [hunk 4]
```diff
@@ -49,6 +53,8 @@ android {
 }
 
 dependencies {
+    // OneSignal SDK
+    implementation 'com.onesignal:OneSignal:[4.0.0, 4.99.99]'
```
Comment (z3bi) [inline app/build.gradle]: I think we can just specify an exact version and then update if needed

File: app/build.gradle [hunk 5]
```diff
@@ -5,6 +5,10 @@ plugins {
     id 'dagger.hilt.android.plugin'
 }
 
+plugins {
+    id 'com.onesignal.androidsdk.onesignal-gradle-plugin'
```
Comment (z3bi) [inline app/build.gradle:6]: lets just put this in the plugins section we have above instead of making a new plugins section

**Review by z3bi — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (aahmad4) [reply build.gradle]: How would we do this?

Comment (z3bi) [reply build.gradle]: replace `[0.12.10, 0.99.99]` with just `0.12.10`

Comment (z3bi) [reply build.gradle]: actually it looks like the latest is `0.14.0` see https://github.com/OneSignal/OneSignal-Gradle-Plugin/releases

File: app/build.gradle (hunk 4 above)
Comment (z3bi) [reply app/build.gradle]: and here we can specify `4.6.7` see https://github.com/OneSignal/OneSignal-Android-SDK/releases

File: app/src/main/java/org/raleighmasjid/iar/MainApplication.kt (hunk 2 above)
Comment (z3bi) [reply app/src/main/java/org/raleighmasjid/iar/MainApplication.kt:8]: just be careful, this repo is public so we don't want to commit sensitive keys to the repo. I don't think this app id is that sensitive. In terms of organization I would suggest maybe using a BuildConfig https://guides.codepath.com/android/Storing-Secret-Keys-in-Android#hidden-in-buildconfigs

**Review by z3bi — APPROVED** [review]

**Pull request merged**